library/control/test/ProductFeatures_test.rb

```diff
@@ -158,4 +158,95 @@
       end
     end
   end
+
+  describe "#GetFeature" do
+    let(:scr_root_dir) { File.join(File.dirname(__FILE__), "data") }
+    let(:normal_stage) { false }
+    let(:firstboot_stage) { false }
+
+    before do
+      allow(Yast::Stage).to receive(:normal).and_return(normal_stage)
+      allow(Yast::Stage).to receive(:firstboot).and_return(firstboot_stage)
+    end
+
+    around do |example|
+      change_scr_root(scr_root_dir, &example)
+    end
+
+    context "when features are not initialized yet" do
+      it "initializes them" do
+        expect(subject.GetFeature("globals", "base_product_license_directory")).to_not be_nil
+      end
+    end
+
+    context "when features are already initialized" do
+      before do
+        subject.InitFeatures(true)
+      end
+
+      it "does not force to initialize them again" do
+        expect(subject).to receive(:InitFeatures).with(false)
+
+        subject.GetFeature("globals", "base_product_license_directory")
+      end
+
+      context "but it is in normal stage" do
+        let(:normal_stage) { true }
+
+        it "reads the value from the running system" do
+          # value readed from data/etc/YaST2/ProductFeatures file
+          expect(subject.GetFeature("globals", "base_product_license_directory"))
+            .to eq("/usr/share/licenses/product/base")
+        end
+      end
+
+      context "but it is in firstboot stage" do
+        let(:firstboot_stage) { true }
+
+        it "reads the value from the running system" do
+          # value readed from data/etc/YaST2/ProductFeatures file
+          expect(subject.GetFeature("globals", "base_product_license_directory"))
+            .to eq("/usr/share/licenses/product/base")
```

Comment:
I guess the default in https://github.com/yast/yast-yast2/blob/2fe30e8a1435a14da21a9f11dee5d8ceb59979e0/library/control/src/modules/ProductFeatures.rb#L74 will be later changed from `/etc/YaST2/licenses/base/` to `/usr/share/licenses/product/base`. In that case we cannot easily decide if the value was really read from the system. I'd probably use some custom value in the testing data/etc/YaST2/ProductFeatures file just for the testing purposes, so we can be sure it's really different than the default.

Reply:
> will be later changed from /etc/YaST2/licenses/base/ to /usr/share/licenses/product/base

I am not sure, because the value has changed for some products, but not for all of them. So, shall we really change the current value? I don't think so.

Take a look to https://jira.suse.de/browse/SLE-4173 and https://build.suse.de/request/show/185130

What do you think?

Reply:
> I'd probably use some custom value in the testing data/etc/YaST2/ProductFeatures file just for the testing purposes

Ok, I'll change to a custom value.

Reply:
> I am not sure, because the value has changed for some products, but not for all of them. So, shall we really change the current value? I don't think so.

Ugh, really? Which product is missing? I'd suppose the same value would be used for all SLE15-SP1 product family. Maybe it is a bug. Ask Behlert if that's correct.

Reply:
As we were talking in IRC, TW, Leap and others. Not SLE15-SP1 family.